products/views.py

```diff
@@ -1,3 +1,25 @@
-from django.shortcuts import render
+from django.http  import JsonResponse
+from django.views import View
 
-# Create your views here.
+from .models import Product
+
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        try:
+            product = Product.objects.get(id=product_id)
+
+            images  = product.productimage_set.all()
+            options = product.options.all()
+
+            result = {
+                'id'      : product.id,
+                'name'    : product.name,
+                'price'   : product.price,
+                'is_green': product.is_green,
+                'is_best' : product.is_best,
+                'img_urls': [image.url for image in images],
+                'options' : [option.name for option in options]
```

Comment:
list comprehension으로 구현하셨군요👍

client (frontend)에게 데이터를 Return 할 때에는 id(pk)값을 같이 반환해주는 것이 좋습니다.
```suggestion
                'images': [{
                    'id'  : image.id,
                    'url' : image.url 
                    } for image in images],
                'options' : [{
                    'id'   : option.id,
                    'name' : option.name
                    } option in options]
```

Reply:
감사합니다 수정했습니다~